### ISSUE TITLE
I can't load in any of my models

### ISSUE THREAD

**Issue by Skullcrusher896** [DaanBroekhof/JoroDox#1]
For some reason my models I've exported from C4D take forever to load it. I haven't been able to convert a single model into a mesh

**Comment by Spacemike** [comment]
Any idea when this will be added in to the main build or such? I tried using the react-rewire and it just booted up to just a basic version of the UI where it says "Home page" and, underneath it, it just says "It's just a start..." and clicking on the folder icon just yields "Type: directory" with "C:\" about is and two icons.

**Comment by DaanBroekhof** [comment]
_this issue seems to contain separate issues/people? Perhaps people are confusing this with a general ticket?_

@Spacemike Yeah the react-rewrite branch is very unstable atm. It has some model import/export improvements over the google chrome extension, but not a lot.
Which feature were you specifically looking for?

**Comment by Spacemike** [comment]
Ah, alrighty, and I was specifically looking for the part where it could load C4D models with bones, as the models cinema 4D exports without having any bones will open perfectly and will port perfectly into the game, however, if you add even a single bone to your scene and rig that bone to an object and export it, the JoroDox Tool will just sit there with a spinning gear icon and never load, but if I import it into blender and then export it again, it will load it just fine, however, blender doesn't seem to import the bones right and the positions of all the objects that were attached to bones gets messed up, so I am not sure if it is the way C4D exports it or what, but, then again, based on my experience, trying to import .dae files from one system to another tends to break them.
Here is a picture of what the tool shows when I try to load the original:
![original](https://user-images.githubusercontent.com/8691038/35900699-535d3434-0ba1-11e8-8fda-63bea34e5223.PNG)
Here is what it shows after importing it into blender and then exporting it again:
![after_blender_export](https://user-images.githubusercontent.com/8691038/35900698-5349b152-0ba1-11e8-88c2-299e1e3055f7.PNG)

I also took a look through the coding differences between the two files and most of the differences seemed to only involve naming, and due to my lack of knowledge on how your tool works internally I cannot say what the issue is. However, I can provide exports from the programs, I have attached a zip file that contains 2 sets of .dae files, the first is simply the two .dae files in the images above, the second set is going through the process of attaching a bone to a simple cube, which I have found out that the moment I bind the cube to the bone the tool will no longer load it, if you need me to make more .dae exports to help you figure out why it refuses to read C4D's version of the file I would be more than happy to help! (Also, sorry for the slow response, had a long day with SpaceX's Falcon Heavy launch and the traffic getting back from it was a nightmare)

[collada files.zip](https://github.com/DaanBroekhof/JoroDox/files/1701918/collada.files.zip)

**Comment by DaanBroekhof** [comment]
Hmm yeah Collada (.dae) has a few ways of defining bones - and the package I am using to parse Collada doesn't always pick them up right. I have not tried importing Collada files created from Cinema4D before, so there probably is an incompatibility there. A spinning cog usually means there was an error during import, and I can see that when I dig deeper a bit.

The main branch of this tool (the Chrome extension) is quite out of date, last updated about 2 years ago, but in my experimental branch I have upgraded the Collada parsing package. I will see if that helps with your models and get back to you!

**Comment by Spacemike** [comment]
yeah, I noticed that when comparing blender and cinema 4d's, it's probably also why trying to transfer models in between them never seems to work quite right, I'm not sure why there's just not some universal model file like what Collada has, except it was actually universal and you could port back and forth with ease, but oh well, maybe that will come true some day!

And, I kinda noticed that, although, the actual extension in the chrome extension browser said it was updated a year ago, so I wasn't completely sure which one was the correct one, but I did have a feeling that the spinning gear was a sign of an error, and that you probably didn't have any examples of C4D files to test out with the extension when you were making it.

and, thanks! I really appreciate you making this tool to be able to use! it's much simpler than the complex Maya mess, especially with the fact that you'd have to make all the bones and such in Maya in the first place, so a virtually universal tool like this is a big help to modders of all of the games that run on the Clausewitz Engine!
